### PR TITLE
Create Components anchor

### PR DESCRIPTION
**Pull request by tobias-mintlify**
## Summary

Moves the component docs into their own anchor. I will further update our docs in a future PR so the main docs page is exclusively guides and all reference content lives inside an anchor.

Using RequestExample will hide the code blocks on mobile so I won't merge this in until can add mobile support tomorrow.

## Test Plan

Use `mintlify dev` to preview the changes locally.

### CODE REVIEW

**Comment by hahnbeelee** [comment]
> I will further update our docs in a future PR so the main docs page is exclusively guides and all reference content lives inside an anchor.

What else do you think will need to be moved?

**Comment by tobias-mintlify** [comment]
@hahnbeelee some of the API docs can be cleaned up. We introduce "Param", "ResponseExample", and others alongside the basic concept of an API playground. It would be nice to have guides with clear examples and a link to a separate section with all the components.

**Comment by hahnbeelee** [comment]
@tobias-mintlify brilliant

File: docs/components/overview.mdx [hunk 1]
```diff
@@ -5,6 +5,11 @@ description: "Introduction on how to add and edit content"
...
 Components are reusable building blocks for your documentation. They can be added **directly within your files**. Some components are rendered from [Markdown syntax](https://www.markdownguide.org/basic-syntax/), while other components work the same way as [React components](https://reactjs.org/docs/components-and-props.html).
 
+<Tip>
+  Mintlify provides a library of [built-in components](/components-reference) to
+  make your docs beautiful without formatting things yourself.
+</Tip>
+
 Visit [our text guide](/components/text) to learn basic markdown syntax.
 
 ## Examples
```
Comment (hahnbeelee) [inline docs/components/overview.mdx:15]: If you scroll down there's this section. I feel like you should mention the component reference again here
image

File: docs/mint.json [hunk 2]
```diff
@@ -26,6 +26,11 @@
     "url": "https://mintlify.com/start"
   },
   "anchors": [
+    {
+      "name": "Components",
+      "icon": "code",
+      "url": "components-reference"
```
Comment (hanywang2) [inline docs/mint.json:32]: Could we name the folder `components`? seems cleaner

**Review by hahnbeelee — CHANGES_REQUESTED** [review]
Can you make the 1 change i requested above?

Overall I think the components anchor looks great! I know this is just the first iteration so I would love to see more references to the components anchor from the main anchor because even if it's great if no one can find it it loses value.

**Review by hahnbeelee — CHANGES_REQUESTED** [review]
We need at least 1 reference to the api components from the api playground section in the main section of the docs

**Review by hahnbeelee — APPROVED** [review]

**Pull request merged**